src/common/state/redis.c

```diff
@@ -262,15 +415,48 @@ void redis_object_table_subscribe(table_callback_data *callback_data) {
 }
 
 /*
- *  ==== task_log callbacks ====
+ *  ==== task_table callbacks ====
  */
 
-void redis_task_log_publish(table_callback_data *callback_data) {
+void redis_task_table_get_task_callback(redisAsyncContext *c,
+                                        void *r,
+                                        void *privdata) {
+  REDIS_CALLBACK_HEADER(db, callback_data, r)
+  redisReply *reply = r;
+  /* Check that we received a Redis hashmap. */
+  if (reply->type != REDIS_REPLY_ARRAY) {
+    LOG_ERR("Expected Redis array, received type %d %s", reply->type,
+            reply->str);
+    exit(-1);
+  }
+  /* If the user registered a success callback, construct the task object from
+   * the Redis reply and call the callback. */
+  if (callback_data->done_callback) {
+    task_table_get_callback done_callback = callback_data->done_callback;
+    task *task_reply = parse_redis_task_table_entry(
+        callback_data->id, reply->elements, reply->element);
+    done_callback(task_reply, callback_data->user_context);
+    free_task(task_reply);
+  }
+  destroy_timer_callback(db->loop, callback_data);
+}
+
+void redis_task_table_get_task(table_callback_data *callback_data) {
+  CHECK(callback_data);
+  db_handle *db = callback_data->db_handle;
+  redisAsyncCommand(db->context, redis_task_table_get_task_callback,
+                    (void *) callback_data->timer_id, "HGETALL task:%b",
+                    callback_data->id.id, (size_t) UNIQUE_ID_SIZE);
+}
+
+void redis_task_table_publish(table_callback_data *callback_data,
```

Comment:
It's possible that later we will want to make the HMSET and the PUBLISH atomic using MULTI/EXEC.. (just thinking about the global scheduler implementation), but that should be possible if necessary down the road.